crates/rspack_core/src/options/filename.rs

```diff
@@ -215,6 +216,10 @@ impl<F: LocalFilenameFn> Filename<F> {
   }
 }
 
+fn replace_all<'a>(regex: &Lazy<Regex>, haystack: &'a str, rep: &str) -> Cow<'a, str> {
+  regex.replace_all(haystack, rep.split("$").join("$$"))
```

Comment:
Can this be optimized based on its call site? I believe this can be omitted with replacers without `$` in it and calling `Itertools::join` creates extra overhead of `String`s.

Reply:
Seems can use NoExpand()